src/main.rs

```diff
@@ -282,11 +269,29 @@ fn main() {
         }
     }
 
```

Comment:
We could use some comments here about what's going on with the DNS, registration and tunneling servers.